includes/class-wc-subscription-data-copier.php

```diff
@@ -0,0 +1,463 @@
+	 * @return array The data to be copied with the excluded keys removed.
+	 */
+	private function filter_excluded_meta_keys_via_query( $data ) {
+		$excluded_keys = $this->get_excluded_data_keys();
+
+		foreach ( $data as $meta_key => $meta_value ) {
+			if ( isset( $excluded_keys['in'] ) && in_array( $meta_key, $excluded_keys['in'], true ) ) {
+				unset( $data[ $meta_key ] );
+			} elseif ( isset( $excluded_keys['regex'] ) ) {
+				foreach ( $excluded_keys['regex'] as $regex ) {
+					if ( preg_match( $regex, $meta_key ) ) {
+						unset( $data[ $meta_key ] );
+						break;
+					}
+				}
+			}
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Returns the deprecated meta database query that returns the "from" objects meta data.
+	 *
+	 * Triggers a deprecation notice if the deprecated "wcs_{$this->copy_type}_meta_query" filter is in use by at least 1 third-party.
+	 *
+	 * @return string SQL SELECT query.
+	 */
+	private function get_deprecated_meta_query() {
+		global $wpdb;
+
+		$meta_query = sprintf(
+			"SELECT `meta_key`, `meta_value`
+			 FROM %s
+			 WHERE `post_id` = %d
+			 AND `meta_key` NOT LIKE '%s'
+			 AND `meta_key` NOT IN ('%s')",
+			$wpdb->postmeta,
+			$this->from_object->get_id(),
+			'_schedule_%',
+			implode( "', '", self::DEFAULT_EXCLUDED_META_KEYS )
+		);
+
+		if ( in_array( $this->copy_type, array( 'renewal_order', 'parent' ), true ) ) {
+			$meta_query .= " AND `meta_key` NOT LIKE '_download_permissions_granted' ";
+		}
+
+		if ( $this->has_filter_on_meta_query_hook() ) {
+			wcs_deprecated_hook( "wcs_{$this->copy_type}_meta_query", '2.0.0', "wcs_{$this->copy_type}_meta" );
+			$meta_query = apply_filters( "wcs_{$this->copy_type}_meta_query", $meta_query, $this->to_object, $this->from_object );
+		}
+
+		return $meta_query;
+	}
+
+	/**
+	 * Applies the deprecated "wcs_{$this->copy_type}_meta filter.
+	 *
+	 * Triggers a deprecation notice if the deprecated "wcs_{$this->copy_type}_meta" filter is in use by at least 1 third-party.
+	 *
+	 * @param array $data The data to copy.
+	 * @return array The filtered set of data to copy.
+	 */
+	private function apply_deprecated_filter( $data ) {
+		// Only continue if the filter is use.
+		if ( ! has_filter( "wcs_{$this->copy_type}_meta" ) ) {
+			return $data;
+		}
+
+		// Convert the data into the backwards compatible format ready for filtering - wpdb's ARRAY_A format.
+		$data_array = array();
+
+		foreach ( $data as $key => $value ) {
+			$data_array[] = array(
+				'meta_key'   => $key, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key -- This is a meta key, not a query.
+				'meta_value' => $value, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value -- This is a meta value, not a query.
+			);
+		}
+
+		wcs_deprecated_hook( "wcs_{$this->copy_type}_meta", '2.0.0', "wcs_{$this->copy_type}_meta_data" );
+		$data_array = apply_filters( "wcs_{$this->copy_type}_meta", $data_array, $this->to_object, $this->from_object );
```

Comment:
I decided to deprecate this filter for 2 reasons. 

1. It use to filter data in the format:

```php
array
(
    [0] => Array
        (
            [meta_key] => _customer_user
            [meta_value] => 1
        )

    [1] => Array
        (
            [meta_key] => _cart_hash
            [meta_value] => 811128b572bcfca774a148c90a2736fb
        )
...
```
This is rather unhelpful as callbacks would need to loop over each piece of data and check the key. The new filter will enable the much more efficient `unset( 'custome_meta_key' )`. 

2. The `_meta` suffix on this filter hook is technically incorrect in a HPOS world. Previously this filter did filter meta data - the data stored in the post meta table. This is no longer correct in HPOS environments. It filters all data including operational, address and meta data.